src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -1494,52 +1488,51 @@ internal Type[] FunctionPointerReturnAndParameterTypes
+                    : null);
+
+            private static bool IsFullNameRoundtripCompatible(RuntimeType runtimeType)
             {
-                switch (kind)
-                {
-                    case TypeNameKind.Name:
-                        // No namespace, full instantiation, and assembly.
-                        return ConstructName(ref m_name, TypeNameFormatFlags.FormatBasic);
-
-                    case TypeNameKind.FullName:
-                        // We exclude the types that contain generic parameters because their names cannot be roundtripped.
-                        // We allow generic type definitions (and their refs, ptrs, and arrays) because their names can be roundtriped.
-                        // Theoretically generic types instantiated with generic type definitions can be roundtripped, e.g. List`1<Dictionary`2>.
-                        // But these kind of types are useless, rare, and hard to identity. We would need to recursively examine all the
-                        // generic arguments with the same criteria. We will exclude them unless we see a real user scenario.
-                        if (!m_runtimeType.GetRootElementType().IsGenericTypeDefinition && m_runtimeType.ContainsGenericParameters)
-                            return null;
-
-                        // Exclude function pointer; it requires a grammar update and parsing support for Type.GetType() and friends.
-                        // See https://learn.microsoft.com/dotnet/framework/reflection-and-codedom/specifying-fully-qualified-type-names.
-                        if (m_runtimeType.IsFunctionPointer)
-                            return null;
-
-                        // No assembly.
-                        return ConstructName(ref m_fullname, TypeNameFormatFlags.FormatNamespace | TypeNameFormatFlags.FormatFullInst);
-
-                    case TypeNameKind.ToString:
-                        // No full instantiation and assembly.
-                        return ConstructName(ref m_toString, TypeNameFormatFlags.FormatNamespace);
-
-                    default:
-                        throw new InvalidOperationException();
-                }
+                // We exclude the types that contain generic parameters because their names cannot be round-tripped.
+                // We allow generic type definitions (and their refs, ptrs, and arrays) because their names can be round-tripped.
+                // Theoretically generic types instantiated with generic type definitions can be round-tripped, e.g. List`1<Dictionary`2>.
+                // But these kind of types are useless, rare, and hard to identity. We would need to recursively examine all the
+                // generic arguments with the same criteria. We will exclude them unless we see a real user scenario.
+                Type rootElementType = runtimeType.IsPointer ? runtimeType : runtimeType.GetRootElementType();
```

Comment:
Why is the `IsPointer` check necessary here?

Reply:
Current behavior:
```csharp
"System.Void()*" == typeof(delegate*<void>*)
"System.Void()**" == typeof(delegate*<void>**)
```

Without the `IsPointer` check:
```csharp
null == typeof(delegate*<void>*)
null == typeof(delegate*<void>**)
```

I needed to update the tests after I checked some existing behavior locally.

Reply:
It would mean that `ContainsGenericParameters` returns true for a function pointer without any generic parameters. Sounds like a bug somewhere else?

Reply:
> "System.Void()*" == typeof(delegate*<void>*)

The current behavior is not incorrect. `System.Void()*` is not a roundtripable name. You can pass it to `Type.GetType` and it is going to fail.